### PR TITLE
Implemented redesign of dashboard page

### LINKED ISSUE #766: Dashboard: New UI/UX to match current designs
As seen in [MM2 on Indesign](https://invis.io/9B804AMXA#/172619323_Dashboard)

dashboard 2x

- [x] update to current header
- [ ] only implement "dashboard" in the top nav for now
- [x] Update the user widget
- [ ] Update the courses widget (except for current grade, and pricing)
- [ ] Add tool tip widget, to explain "verified" See https://projects.invisionapp.com/share/H486ZHFJ3#/screens
- [ ] Fix #780 Dashboard looks different in FireFox

This issue does not include:

- Grading curve widget
- Learners near you widget
- Progress widget
- My Groups widget
- Current grades in t

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #766 

#### What's this PR do?
Changes dashboard UI code to match mockups on #766, with the exception of certain cases mentioned in #833 

#### Where should the reviewer start?
static/js/components/dashboard/CourseListCard.js

#### How should this be manually tested?
Get a bunch of interesting runs and courses to display in your dashboard, or stub out `getDashboard` in `api.js` like this:
    import { DASHBOARD_RESPONSE } from '../constants';
    export function getDashboard() {
      return Promise.resolve(DASHBOARD_RESPONSE);
    }

Look carefully at the mockups and the course descriptions and verify that all cases match up as expected.

#### Any background context you want to provide?
All programs are shown in this PR, one per card. It looks like the mockup is for a single program but I'm not sure where the user specifies this.

#### Screenshots (if appropriate)

screenshot from 2016-08-11 17-47-02
screenshot from 2016-08-11 17-47-11

### CODE REVIEW

**Comment by aliceriot** [comment]
I think overall the UI looks great! There might be a bit more `margin-top` than in the mockup, but that's the only layout difference I see.

**Comment by aliceriot** [comment]
looking at code now

**Comment by giocalitri** [comment]
@noisecapella I think there is a tiny linting problem

**Comment by aliceriot** [comment]
@giocalitri I think this is still pretty WIP

**Comment by giocalitri** [comment]
ops... I did not notice the label: sorry!

File: static/js/components/dashboard/CourseGrade.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const { course } = this.props;
+    let firstRun = {};
+    if (course.runs.length > 0) {
+      firstRun = course.runs[0];
+    }
+
+    let percent = null;
+    let description = null;
+
+    if (firstRun.grade !== undefined) {
```
Comment (aliceriot) [inline static/js/components/dashboard/CourseGrade.js]: is there a grade attribute on `firstRun` if `course.runs.length` is not `>0`?

Comment (aliceriot) [inline static/js/components/dashboard/CourseGrade.js]: Oh I see, in that case we'd just end up putting an empty `{percent}` and `{description}` on the page. nvm, carry on.

File: static/js/components/dashboard/CourseDescription.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      }
+    } else if (course.status === STATUS_VERIFIED_NOT_COMPLETED) {
+      if (firstRun.course_start_date) {
+        let courseStartDate = moment(firstRun.course_start_date);
+        if (courseStartDate.isAfter(now, 'day')) {
+          let formattedDate = courseStartDate.format(DASHBOARD_FORMAT);
+          text = `Begins ${formattedDate}`;
+        }
+      }
+    }
```
Comment (aliceriot) [inline static/js/components/dashboard/CourseDescription.js]: what do you think of using a `switch` statement instead for this big conditional here?

File: static/js/components/dashboard/CourseDescription.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  STATUS_NOT_PASSED,
+  STATUS_ENROLLED_NOT_VERIFIED,
+  STATUS_VERIFIED_NOT_COMPLETED,
+  DASHBOARD_FORMAT,
+} from '../../constants';
+
+export default class CourseDescription extends React.Component {
+  props: {
+    course: Course,
+    now: Object,
```
Comment (aliceriot) [inline static/js/components/dashboard/CourseDescription.js]: `now` is a momentjs object right? We have flow type annotations installed for moment.

File: static/js/components/dashboard/CourseAction.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      action = <span>
+        <Button
+          className="mm-button-action dashboard-button"
+          disabled={disabled}
+          {...linkProps}
+        >
+          Enroll
+        </Button>
+        <span className="sr-only"> in {firstRun.title}</span>
+      </span>;
```
Comment (aliceriot) [inline static/js/components/dashboard/CourseAction.js]: this is very close to lines 35-40, what do you think about pulling this out into a little helper function?

File: static/js/components/dashboard/CourseRow.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    course: Course,
+    now: Object
+  };
+
+  render() {
+    const { course, now } = this.props;
+
+    return <Grid className="course-row">
+      <Cell col={6}>
+        <CourseDescription course={course} now={now} />
```
Comment (aliceriot) [inline static/js/components/dashboard/CourseRow.js]: looks like you could just splat `{...this.props}` into all these course components here, maybe it's better to be explicit tho?

**Comment by aliceriot** [comment]
@noisecapella I had a couple of questions, but generally this looks good!

File: static/js/components/dashboard/CourseDescription.js (hunk 2 above)
Comment (noisecapella) [inline static/js/components/dashboard/CourseDescription.js]: Sure

File: static/js/components/dashboard/CourseDescription.js (hunk 3 above)
Comment (noisecapella) [inline static/js/components/dashboard/CourseDescription.js]: Yes, I'll address this after the new flow PR is merged

File: static/js/components/dashboard/CourseRow.js (hunk 5 above)
Comment (noisecapella) [inline static/js/components/dashboard/CourseRow.js]: Yeah, I'd prefer to be explicit especially when there aren't that many props to forward

**Comment by noisecapella** [comment]
Should be ready for review again!

**Comment by aliceriot** [comment]
I'm getting some sort of font-related error:

font

**Comment by aliceriot** [comment]
functionality looks great otherwise!

**Comment by noisecapella** [comment]
It looks like the 600 font weight isn't supported for Roboto so I removed it and added 400i which we use in this PR. How does it look now?

**Comment by aliceriot** [comment]
something is still wacky: 

more-font

**Comment by noisecapella** [comment]
Removed a duplicate fonts URL

**Comment by aliceriot** [comment]
that fixed it!

File: static/js/components/dashboard/CourseDescription.js (hunk 3 above)
Comment (aliceriot) [inline static/js/components/dashboard/CourseDescription.js]: bump!

File: static/js/components/dashboard/CourseAction.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  DASHBOARD_FORMAT,
+} from '../../constants';
+
+export default class CourseAction extends React.Component {
+  props: {
+    course: Course,
+    now: Object,
+  };
+
+  makeActionButton = (text, run, disabled, url) => {
```
Comment (aliceriot) [inline static/js/components/dashboard/CourseAction.js]: flow is complaining about this method not being annotated

**Comment by aliceriot** [comment]
Code mostly looks good to me!

Only thing is the `now` objects, I think we should update them to be of `moment$Moment` type instead of `Object`, and one flow error.

File: static/js/components/dashboard/CourseDescription_test.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+  STATUS_NOT_PASSED,
+  STATUS_OFFERED_NOT_ENROLLED,
+  STATUS_ENROLLED_NOT_VERIFIED,
+  STATUS_VERIFIED_NOT_COMPLETED,
+  STATUS_NOT_OFFERED,
+  ALL_COURSE_STATUSES,
+} from '../../constants';
+
+
+export let findCourse = courseSelector => {
```
Comment (aliceriot) [inline static/js/components/dashboard/CourseDescription_test.js]: flow is complaining about this function too

**Comment by noisecapella** [comment]
Ok, flow checks out now

**Comment by aliceriot** [comment]
LGTM +1

**Pull request merged**